application/src/main/java/org/togetherjava/tjbot/commands/filesharing/FileSharingMessageListener.java

```diff
@@ -0,0 +1,183 @@
+package org.togetherjava.tjbot.commands.filesharing;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import net.dv8tion.jda.api.entities.ChannelType;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.ThreadChannel;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.interactions.components.buttons.Button;
+import org.jetbrains.annotations.NotNull;
+import org.togetherjava.tjbot.commands.MessageReceiverAdapter;
+import org.togetherjava.tjbot.config.Config;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+/**
+ * Listener that receives all sent help messages and uploads them to a share service if the message
+ * contains a file with the given extension in the
+ * {@link FileSharingMessageListener#extensionFilter}
+ */
+public class FileSharingMessageListener extends MessageReceiverAdapter {
+
+    private static final ObjectMapper JSON = new ObjectMapper();
+
+    private final String gistApiKey;
+    private static final String SHARE_API = "https://api.github.com/gists";
+    private static final HttpClient CLIENT = HttpClient.newHttpClient();
+
+    private final Set<String> extensionFilter = Set.of("txt", "java", "gradle", "xml", "kt", "json",
+            "fxml", "css", "c", "h", "cpp", "py", "yml");
+
+    private final Predicate<String> isStagingChannelName;
+    private final Predicate<String> isOverviewChannelName;
+
+
+    public FileSharingMessageListener(@NotNull Config config) {
+        super(Pattern.compile(".*"));
+
+        gistApiKey = config.getGistApiKey();
+        isStagingChannelName = Pattern.compile(config.getHelpSystem().getStagingChannelPattern())
+            .asMatchPredicate();
+        isOverviewChannelName = Pattern.compile(config.getHelpSystem().getOverviewChannelPattern())
+            .asMatchPredicate();
+    }
+
+    @Override
+    public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+        User author = event.getAuthor();
+        if (author.isBot() || event.isWebhookMessage()) {
+            return;
+        }
+
+        if (!isHelpThread(event)) {
+            return;
+        }
```

Comment:
u don't need this check, see #512

Reply:
I think I need it, cause it should only upload it if the file is in a help thread

with this check:
![image](https://user-images.githubusercontent.com/88111627/184533035-2315eea2-dbca-4264-9881-0f4e800514a4.png)
![image](https://user-images.githubusercontent.com/88111627/184533094-2fb7e6f2-ae9d-4127-81a8-126b7d6cd8b5.png)

without:
![image](https://user-images.githubusercontent.com/88111627/184533066-7ee6cd55-524c-425f-b8da-b72cb71418a1.png)
![image](https://user-images.githubusercontent.com/88111627/184533076-65e28db0-b562-4a4b-8977-8904c725b8a2.png)

Reply:
ohh yea, sry i misunderstood it